internal/storage/payload_reader.go

```diff
@@ -80,18 +80,27 @@ func (r *PayloadReader) GetBoolFromPayload() ([]bool, error) {
 	if r.colType != schemapb.DataType_Bool {
 		return nil, fmt.Errorf("failed to get bool from datatype %v", r.colType.String())
 	}
-	reader, ok := r.reader.RowGroup(0).Column(0).(*file.BooleanColumnChunkReader)
-	if !ok {
-		return nil, fmt.Errorf("expect type *file.BooleanColumnChunkReader, but got %T", r.reader.RowGroup(0).Column(0))
-	}
 
+	var offset int
 	values := make([]bool, r.numRows)
-	total, valuesRead, err := reader.ReadBatch(r.numRows, values, nil, nil)
-	if err != nil {
-		return nil, err
+	for i := 0; i < r.reader.NumRowGroups(); i++ {
+		rg := r.reader.RowGroup(i)
+		for j := 0; j < rg.NumColumns(); j++ {
+			reader, ok := rg.Column(j).(*file.BooleanColumnChunkReader)
```

Comment:
you can not iterate through all columns because there is no guarantee that they are the same data type

Reply:
Fixed